addons/product/models/product.py

```diff
@@ -745,7 +754,8 @@ class ProductPackaging(models.Model):
     company_id = fields.Many2one('res.company', 'Company', index=True)
 
     _sql_constraints = [
-        ('positive_qty', 'CHECK(qty > 0)', 'Contained Quantity should be positive.')
+        ('positive_qty', 'CHECK(qty > 0)', 'Contained Quantity should be positive.'),
+        ('barcode_uniq', 'unique(barcode)', 'A barcode can only be assigned to one packaging.'),
```

Comment:
There was previously a concern about if this would break a database during an upgrade: https://github.com/odoo/upgrade/pull/1703#issuecomment-706970469 I'm not sure if only warning the customer is still the best/only option. It probably is, but we should double check with the upgrade guys.

Reply:
Indeed
I have updated odoo/upgrade#3180, see _[IMP] product/saas~15.2: uniqueness of barcodes_